modules/gapi/src/backends/onnx/gonnxbackend.cpp

```diff
@@ -185,10 +189,21 @@ inline std::vector<int> toCV(const std::vector<int64_t> &vsz) {
 }
 
 inline cv::Mat toCV(Ort::Value &v) {
-    auto info = v.GetTensorTypeAndShapeInfo();
-    return cv::Mat(toCV(info.GetShape()),
-                   toCV(info.GetElementType()),
-                   reinterpret_cast<void*>(v.GetTensorMutableData<uint8_t*>()));
+    const auto info = v.GetTensorTypeAndShapeInfo();
+    const auto type = toCV(info.GetElementType());
+    const auto shape = toCV(info.GetShape());
+    if (type != -1) {
+        return cv::Mat(shape,
+                       type,
+                       reinterpret_cast<void*>(v.GetTensorMutableData<uint8_t*>()));
+    }
+    const int total = std::accumulate(shape.begin(), shape.end(), 0);
+    const int64_t* ptr = v.GetTensorMutableData<int64_t>();
+    cv::Mat mt({total}, CV_32S);
```

Comment:
Why don't you create `cv::Mat` with correct shape ? 
`cv::Mat mat(shape, CV_32S)`

Reply:
For instance:

```
std::transform(ptr, ptr + total, mat.data, [](int64_t v) { return static_cast<int>(v); }
```

Reply:
Added transform.